django/core/management/commands/makemigrations.py

```diff
@@ -211,9 +221,11 @@ def write_migration_files(self, changes):
                         migration_string = writer.path
                     if migration_string.startswith('..'):
                         migration_string = writer.path
-                    self.stdout.write('  %s\n' % self.style.MIGRATE_LABEL(migration_string))
+                    self.log.write('  %s\n' % self.style.MIGRATE_LABEL(migration_string))
                     for operation in migration.operations:
-                        self.stdout.write('    - %s' % operation.describe())
+                        self.log.write('    - %s' % operation.describe())
+                    if self.separate_logs:
```

Comment:
Since `separate_logs` seems like a higher-level mode that does multiple things, maybe you can make it so the mode doesn't need to be stored as an attribute. For example, the following line could write to a `self.output_stream` that defaults to `os.devnull`. When running in script mode, it could be set to `self.stdout`. It would also eliminate the need for an `if` statement.